src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -52,67 +76,165 @@ public void AddDependencyLocation(string fullPath)
                 {
                     _knownAssemblyPathsBySimpleName[simpleName] = paths.Add(fullPath);
                 }
+
+                _ = _analyzerAssemblyInfoMap[fullPath] = null;
             }
         }
 
-        public Assembly LoadFromPath(string fullPath)
+        public Assembly LoadFromPath(string originalAnalyzerPath)
         {
-            CompilerPathUtilities.RequireAbsolutePath(fullPath, nameof(fullPath));
-            return LoadFromPathUnchecked(fullPath);
-        }
+            CompilerPathUtilities.RequireAbsolutePath(originalAnalyzerPath, nameof(originalAnalyzerPath));
+
+            (AssemblyName? assemblyName, string _) = GetAssemblyInfoForPath(originalAnalyzerPath);
+
+            // Not a managed assembly, nothing else to do
```

Comment:
This is not a behavior change, right?

I am thinking of #54899 and I feel like we should permit the analyzer loader to load native dependencies. We know that people are working around this by doing IO in their extensions. It doesn't need to happen in this PR, but we do have it paged in now, and it might be a good time to just do it.

We know that when a native assembly is referenced in the build, that we will enter this path in an attempt to load any analyzers in it. Right? We could make Load take a nullable assemblyName, and require that a native assembly is found when assemblyName is null, and that a managed assembly is found when assemblyName is not null. If a native assembly is found, don't bother searching for analyzers in it, but do register it as a dependency location (which is already done at an earlier phase).

Reply:
> This is not a behavior change, right?

Correct. This threw before as well. It's caught up in `AnalyzerFileReference.AddExtensions` where a diagnostic is registered for the exception. 

> I am thinking of https://github.com/dotnet/roslyn/issues/54899 and I feel like we should permit the analyzer loader to load native dependencies.

Lukewarm agree but it's more complicated than just removing the exception path here though. Have to consider this becomes the mechanism for passing any binary dependency threw to analyzers and that means we likely need to work with IDE to make sure these aren't inspected in ways that would cause issues if it suddenly became a large non-PE file